src/compiler/checker.ts

```diff
@@ -13161,6 +13161,8 @@ namespace ts {
             checkGrammarDecorators(node) || checkGrammarModifiers(node) || checkGrammarProperty(node) || checkGrammarComputedPropertyName(node.name);
 
             checkVariableLikeDeclaration(node);
+
+            checkNameOfStaticMethodOrPropertyInClass(node);

```

Comment:
I would move it to the end of the line  [13161](https://github.com/Microsoft/TypeScript/pull/9215/files#diff-c3ed224e4daa84352f7f1abcd23e8ccaR13161)